tensorflow/lite/micro/kernels/broadcast_to.cc

```diff
@@ -0,0 +1,138 @@
+  }
+  ~BroadcastToContext() {
+    micro_context->DeallocateTempTfLiteTensor(input);
+    micro_context->DeallocateTempTfLiteTensor(shape);
+    micro_context->DeallocateTempTfLiteTensor(output);
+  }
+  MicroContext* micro_context;
+  TfLiteTensor* input;
+  TfLiteTensor* shape;
+  TfLiteTensor* output;
+};
+
+TfLiteStatus ValidateOutputTensor(TfLiteContext* context,
+                                  BroadcastToContext* op_context) {
+  // Ensures the shape is 1D tensor.
+  TF_LITE_ENSURE_EQ(context, NumDimensions(op_context->shape), 1);
+
+  // Ensure output dims is not less than input dims.
+  int input_num_dims = NumDimensions(op_context->input);
+  int output_num_dims = NumDimensions(op_context->output);
+  int shape_num_dims = SizeOfDimension(op_context->shape, 0);
+  TF_LITE_ENSURE_MSG(context, output_num_dims == shape_num_dims,
+                     "Output must match with the expected shape dimension.");
+  TF_LITE_ENSURE_MSG(context, input_num_dims <= output_num_dims,
+                     "Output shape must be broadcastable from input shape.");
+  TF_LITE_ENSURE_MSG(context, output_num_dims <= kMaxDims,
+                     "BroadcastTo only supports 1-5D tensor.");
+
+  // Check if output shape is broadcastable from input shape.
+  auto get_shape_data = [op_context](int i) -> int32_t {
+    if (op_context->shape->type == kTfLiteInt32) {
+      return GetTensorData<int32_t>(op_context->shape)[i];
+    } else {
+      return GetTensorData<int64_t>(op_context->shape)[i];
+    }
+  };
+
+  int extending_dims = output_num_dims - input_num_dims;
+  for (int idx = 0; idx < input_num_dims; ++idx) {
+    TF_LITE_ENSURE_MSG(context,
+                       (SizeOfDimension(op_context->input, idx) == 1 ||
+                        SizeOfDimension(op_context->input, idx) ==
+                            get_shape_data(extending_dims + idx)),
+                       "Output shape must be broadcastable from input shape.");
+  }
+
+  // Validating the shape of the output tensor.
+  tflite::RuntimeShape output_shape =
+      tflite::GetTensorShape(op_context->output);
+  for (int idx = 0; idx < output_num_dims; ++idx) {
+    TF_LITE_ENSURE(context, output_shape.Dims(idx) == get_shape_data(idx));
+  }
+  return kTfLiteOk;
+}
+
+TfLiteStatus Prepare(TfLiteContext* context, TfLiteNode* node) {
+  TF_LITE_ENSURE(context, NumInputs(node) == 2);
+  TF_LITE_ENSURE_EQ(context, NumOutputs(node), 1);
+
+  BroadcastToContext op_context(context, node);
+
+  TF_LITE_ENSURE_MSG(context, (NumDimensions(op_context.input) <= kMaxDims),
+                     "BroadcastTo only supports 1-5D tensor.");
+
+  TF_LITE_ENSURE(context, op_context.shape->type == kTfLiteInt32 ||
+                              op_context.shape->type == kTfLiteInt64);
+  TF_LITE_ENSURE_EQ(context, op_context.input->type, op_context.output->type);
+
+  // Not yet support string type due to the use of memcopy with fixed size.
```

Comment:
s/Strings are unsupported due to their variable size?

Reply:
Yes and this constraint is carried from the lite's implementation: lite does not support it, so no need for micro to support it either I suppose

Reply:
Ack. Perhaps comment can be updated to clear up. I was confused at first.

Reply:
Done. Reword the original verbatim comment copy from tflite to something more specific for micro